### ISSUE TITLE
widgets APP_MODE

### ISSUE THREAD

**Issue by JimB40** [EdgeTX/edgetx#712]
LUA API & UI  naming
Now it's even more clear for me we should go with App Mode naming
Full screen regards screen layout/widget visual output not operational mode and I'm 100% sure it will be confusing.
We can have simple LUA script that display something without need of User interaction via events and it operates in Full screen layout.
So:
- FullScreen Mode - is changing widget layout to full screen that was set by user in different layout
- App Mode - is enabling event sensing to widget

Therefore I propose:
1. Change in widget contextual menu name from Full screen to App mode or Application mode. In C++ this will pass events to selected widget that is in App Mode 
2. Add LUA API widget constant APP_MODE (true/false)
3. Add LUA API functions lcd.widgetEnterFullScreenMode(), lcd.widgetExitFullScreenMode()
4. Pass decision to change widget layout (ie going to full screen) to LUA API function via lcd.widgetEnterFullScreen() function 
5. Change proposed by @jfrickmann function name to lcd.widgetExitAppMode() and change it's functionality to loose event sensing. (APP_MODE = false)

Scenarios:
Entering App Mode
LUA dev can sense that user set widget is in non full screen layout and implement message or kind of preview version or whatever he likes (ie. half screen layout version of App).
Once App Mode command is executed (double tap & widget APP_MODE = true) LUA dev can 
1. proceed with current layout (only one widget is receiving events) and/or provide button to issue lcd.widgetEnterFullScreenMode() if Full Screen mode is available in App
2. issue lcd.EnterFullScreen() and proceed with full screen layout.
Exit App Mode
Once EXIT long is pressed: 
1. If initially widget layout was full screen OS sets APP_MODE = false and LUA dev can sense it  displaying again 'Double tap to start App', 'Dimming display' or something like that but we stay in initial layout (full screen)
2. If initially widget layout was non-full screen OS switches to main view but selected widget stays in APP mode (APP_MODE = true) but now operates in non full screen layout.
 
It may sounds complicated  but opens huge new area for customisation.
Let's say I have main view with four widgets and they are Apps like TBS AL, iNav, Yaapu etc:
1. In main view I'm displaying logos, status information and message to go into App Mode/Full Screen Mode in all of them
2. Once one of them is  double tapped I can 
a. still display small version with let's say settings change possibility 
b. small version with few features and button to go full screen
c. change display to full screen if this is only widget layout provided    

To recap
1. App Mode (APP_MODE = true) is when selected one widget receives events
2. Full Screen Mode is when widget was set by user in non-full layout but App provides full screen layout UI and is displayed in this format
3. Full screen layout is just one of layouts widget dimensions are (480x272 for TX16S  or 320x480 for NV14)

Hope I'm clear with idea.

**Comment by raphaelcoeffic** [comment]
@JimB40 i will have trillions of questions concerning this feature 😅
